### DISCUSSION TITLE
Help with old plugin, target class does not exist

### DISCUSSION THREAD

**Discussion by hibangmike** [wintercms/winter#889] [Q&A] [answered]
Hi, so I’m using a plugin from october v1 and migrate it to my winter 1.2.1 installation using PHP 8.2, the backend works without an issue, it's working fine. I have change all the namespace to use Winter\Storm

however when I load in the frontend, I got this error :

`Target class [octommerce.octommerce.helper] does not exist.`

and here is the code in the facade/Cart.php file
```

<?php namespace Octommerce\Octommerce\Facades;

use Winter\Storm\Support\Facade;

class Cart extends Facade
{
    protected static function getFacadeAccessor()
    {
        return 'octommerce.octommerce.helper';
    }

    protected static function getFacadeInstance()
    {
        return new \Octommerce\Octommerce\Helpers\Cart;
    }
}
```

As I am not proficient in PHP, any help is appreciated
Thank you

**Comment by bennothommo** [comment]
@hibangmike The problem here is that the `getFacadeInstance()` method is no longer called in Laravel 9 - it only uses the `getFacadeAccessor()` to bind a facade to a provider/class using the provider name (`octommerce.octommerce.helper`), and it appears that the Octommerce plugin does not define the `octommerce.octommerce.helper` service anywhere.

What they would need to do is bind the service in the `register()` or `boot()` method of their `Plugin.php` - something like the following:

```php
public function register()
{
    App::bind('octommerce.octommerce.helper', function () {
        return new \Octommerce\Octommerce\Helpers\Cart;
    });
}
```

**Reply by bennothommo** [reply]
Alternatively, you could create your own plugin which does exactly this, if you don't want to edit their plugin directly. This would at least resolve that error you're receiving.

I'd say though, given the plugin was last edited 5 years ago, it's likely it doesn't work with Winter 1.2.

**Comment by hibangmike** [comment]
Hi @bennothommo , Appreciate the reply and also the code, I have tried to use it in the register Plugin.php and also create new plugin, but it's not working, it still got error :

`Class "Octommerce\Octommerce\App" not found `

I agree with what you said, the last edit to the plugin was 5 years ago and it would be plugging holes everywhere to get it to work with Winter 1.2, as it is not worth the time. Right now it is working fine with October, but it still running PHP 7.4, and we want to upgrade to use PHP 8 and migrate to Winter platform.

But I think we are going to leave this project as is. I really appreciate the reply and help you give, Thank you !

**Reply by bennothommo** [reply, marked as answer]
@hibangmike it's worth giving it a try, at least.

That error is simply because I didn't provide you one extra line:

```php
use App;
```

Put that after the `namespace`, where the other `use` cases are. :)

**Reply by LukeTowers** [reply]
I'd say it's highly likely that it still works unless it was doing anything crazy, the backwards compatibility we've maintained has been pretty reliable for the most part.

**Reply by hibangmike** [reply]
Hi @bennothommo ,This is working perfectly, there are also required plugin that have this issue, and I just replicate your code above and change the path. It's working great! right now we are testing thoroughly but it looks like this is really stable enough for us to actually migrate to PHP 8 and finally move to the Winter platform.

Yes @LukeTowers the backend works without any issues, I also change all the namespace to use Winter\Storm even though I have test it before changing all the namespace and it still working flawlessly. 

Thank you so much @bennothommo  for your time and guidance, we will slowly and gradually migrate our other projects to use Winter !

**Reply by bennothommo** [reply]
That's great to hear @hibangmike. Let us know if there's any other dramas and we'll be happy to help out.
